src/dwarf.cpp

```diff
@@ -1366,6 +1366,28 @@ QString Dwarf::happiness_name(DWARF_HAPPINESS happiness) {
     }
 }
 
+QString Dwarf::get_all_goal_names() {
+    if (m_goals.size() > 0) {
+        QHashIterator<UNIT_GOAL_TYPE, short> goals(m_goals);
+
+        QString all_goal_names;
+        while (goals.hasNext())
```

Comment:
I was going to tell you to use a range-based for loop, but they don't actually work very well with QHashMap. This still need a for so use an old-style one: `for (auto goal = m_goals.begin(); goal != m_goals.end(); ++goal)`.

Reply:
Changed to for loop